mono/mini/interp/transform.c

```diff
@@ -6389,6 +6389,12 @@ interp_local_deadce (TransformData *td, int *local_ref_count)
 				interp_clear_ins (td, ins);
 				mono_interp_stats.killed_instructions++;
 			}
+		} else if (MINT_IS_STLOC (ins->opcode) && ins->opcode != MINT_STLOC_VT) {
+			if (!local_ref_count [ins->data [0]] && (td->locals [ins->data [0]].flags & INTERP_LOCAL_FLAG_INDIRECT) == 0) {
+				// We store to a dead stloc, we can replace it with a POP to save local space
```

Comment:
does it really save local space? I thought at this point allocation for locals has already be done, and we can't compact it. do you mean it "saves a store operation"?

Reply:
https://github.com/mono/mono/pull/16714 defers local offset / stack allocation. A local and all its potentially used stack is dead by design if no instruction references it.